logictest: don't skip under stress

tbg opened the pull request:
I'm not sure why we decided to do this in the first place, but it's really
annoying not to be able to stress test logic tests, especially since it
isn't obvious that it's not possible until you waste some time and find out.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/16985)
<!-- Reviewable:end -->

benesch commented:
This has burned both me and Peter, so I'm 👍 in theory—but this _must_ break a nightly job or something. Maybe `teamcity-stress.sh`?

tamird commented:
I think this OOMed the nightlies. @RaduBerinde?

tamird commented:
@RaduBerinde ignore that, doesn't look like you were involved in that issue, but it was https://github.com/cockroachdb/cockroach/issues/10939.

tbg committed:
Makefile: don't pass COCKROACH_STRESS in `make stress`

This was our mechanism to skip `TestLogic` during nightly stress tests.
However it also meant that manual stress testing would skip them. That's
not obvious and also awkward.

Switch back to an environment variable, roughly reverting #14037.

tschottdorf commented:
PTAL at the new changes. Tried to make `Stress()` particular to the nightlies (i.e. not affected by just invoking `make stress`). Did I add the env var to the right script?

benesch commented:
I'm think there's a bit more to do in cmd/teamcity-trigger. I'm just following the git blame trail, though. See https://github.com/cockroachdb/cockroach/pull/14037, which is basically the inverse of this PR.

tamird commented:
LGTM with a squash

tamird commented:
Ah, @benesch is right.

benesch commented:
Yeah, looks like you'll want to flip COCKROACH_STRESS in both teamcity-trigger and github-pull-request-make.

tschottdorf commented:
Thanks for the comments, PTAL!

benesch reviewed (approved):
:tada: thanks for fixing this, @tschottdorf!
